packages/cli/src/commands/site/deploy/handleSiteDeploy.mts

```diff
@@ -17,28 +17,76 @@
 import { consola } from "consola";
 
 import {
+  artifacts,
   ArtifactsSitesAdminV2Service,
   createConjureContext,
   thirdPartyApplicationService,
 } from "#net";
-import type { SiteDeployArgs } from "./SiteDeployArgs.js";
+import archiver from "archiver";
+import * as fs from "node:fs";
+import { Readable } from "node:stream";
+import { ExitProcessError } from "../../../ExitProcessError.js";
+import { autoVersion as findAutoVersion } from "../../../util/autoVersion.js";
+import type { SiteDeployArgs } from "./siteDeployArgs.js";
 
 export default async function handleSiteDeploy(
-  { siteVersion: version, appRid, baseUrl, clearVersion }: SiteDeployArgs,
+  {
+    version,
+    application,
+    foundryUrl,
+    autoVersion,
+    gitTagPrefix,
+    uploadOnly,
+    directory,
+  }: SiteDeployArgs,
 ) {
-  const repositoryRid = await thirdPartyApplicationService
-    .fetchWebsiteRepositoryRid(baseUrl, appRid);
+  if (!version && !autoVersion) {
+    throw new ExitProcessError(
+      2,
+      "Either version or autoVersion must be specified",
+    );
+  }
+
+  const siteVersion = !version ? await findAutoVersion(gitTagPrefix) : version;
+  if (!version) {
+    consola.info(
+      `No version was specified, and autoVersion is enabled. Inferred version: ${siteVersion}`,
+    );
+  }
+
+  const stat = await fs.promises.stat(directory);
+  if (!stat.isDirectory()) {
+    consola.error("Specified path is not a directory");
+    throw new ExitProcessError(2);
+  }
 
-  const ctx = createConjureContext(baseUrl, "/artifacts/api");
-  if (version) {
+  consola.start("Zippping site files");
+
+  const archive = archiver("zip").directory(directory, false);
+
+  await Promise.all([
+    artifacts.SiteAssetArtifactsService.uploadZippedSiteAsset(
```

Comment:
I notice when on errors such as version already exists we also print out the CLI help text which makes it hard to see the error at the bottom do you know where that's coming from?